core/trino-main/src/test/java/io/trino/operator/TestGroupByHash.java

```diff
@@ -502,4 +503,73 @@ public void testMemoryReservationYieldWithDictionary()
         assertEquals(currentQuota.get(), 10);
         assertEquals(currentQuota.get() / 3, yields);
     }
+
+    @Test
+    public void testLowCardinalityDictionaries()
+    {
+        GroupByHash groupByHash = createGroupByHash(
+                TEST_SESSION,
+                ImmutableList.of(BIGINT, BIGINT),
+                new int[] {0, 1},
+                Optional.empty(),
+                100,
+                JOIN_COMPILER,
+                TYPE_OPERATOR_FACTORY,
+                NOOP);
+        Block firstBlock = BlockAssertions.createLongDictionaryBlock(0, 100, 10);
+        Block secondBlock = BlockAssertions.createLongDictionaryBlock(0, 100, 10);
+        Page page = new Page(firstBlock, secondBlock);
+
+        groupByHash.addPage(page).process();
```

Comment:
could you assert that `Work` is of `AddLowCardinalityDictionaryPageWork` type?

Reply:
Done in a separate test

Reply:
> Done in a separate test

Separate test is not sufficient. In that test you test condition, but you don't test result. In this test you test result, but no condition. These tests can also diverge over time so we won't notice that we have untested code paths.

please verify that `Work` is actually of low cardinality type (here and in `testLowCardinalityDictionariesGetGroupIds`).